strawberryfields/compilers/gaussian_merge.py

```diff
@@ -0,0 +1,376 @@
+                if d_gate_successors:
+                    merged_gaussian_ops += d_gate_successors
+
+        # Add gaussian operations that should be exectued at the same time as op
+        # E.X Rgate|q[0] Rgate|q[1] -> BS|q[0]q[1]. Adds Rgate|q[1] if Rgate|q[0] is the op.
+        for gaussian_op in merged_gaussian_ops:
+            for predecessor in self.DAG.predecessors(gaussian_op):
+                if predecessor is op:
+                    continue
+                if (
+                    predecessor not in merged_gaussian_ops
+                    and get_op_name(predecessor) in self.gaussian_ops
+                ):
+                    if self.valid_prepend_op_addition(op, predecessor, merged_gaussian_ops):
+                        merged_gaussian_ops.append(predecessor)
+
+        merged_gaussian_ops = self.remove_invalid_operations(op, merged_gaussian_ops)
+
+        if self.is_redundant_merge(op, merged_gaussian_ops):
+            return []
+        return merged_gaussian_ops
+
+    def is_redundant_merge(self, op, merged_gaussian_ops):
+        """
+        Help function that determines if merge will do nothing. i.e. just contains Gaussian Transforms and Displacement
+        operations.
+        """
+        if "GaussianTransform" in get_op_name(op):
+            all_displacement_gates = True
+            for gate in merged_gaussian_ops:
+                if "Dgate" not in get_op_name(gate):
+                    all_displacement_gates = False
+            if all_displacement_gates:
+                return True
+        return False
+
+    def non_gaussian_qumodes_dependecy(self, op):
+        """
+        Get qumodes used in predecessor non-gaussian operations of op. Returns a list of integers,
+        where the number depicts the qumode index that the non-gaussian operation operates on
+        """
+        for predecessor in self.DAG.predecessors(op):
+            if not self.is_op_gaussian_or_prep(predecessor):
+                return get_qumodes_operated_upon(predecessor)
+        return []
+
+    def organize_merge_ops(self, merged_gaussian_ops):
+        """
+        Organize operations to be merged in order by using the order of the current operation sequence
+        """
+        organized_merge_operations = []
+        for op_in_seq in self.curr_seq:
+            if op_in_seq in merged_gaussian_ops:
+                organized_merge_operations.append(op_in_seq)
+        return organized_merge_operations
+
+    def is_op_gaussian_or_prep(self, op):
+        """
+        Helper function that returns True if op is gaussian or a preparation state else returns False
+        """
+        op_name = get_op_name(op)
+        return op_name in self.gaussian_ops or op_name in self.prep_states
+
+    def valid_prepend_op_addition(self, op, pre, merged_gaussian_ops):
+        """
+        Helper function that ensures predecessor operation being added did not skip any operations between op and pre
```

Comment:
```suggestion
        Helper function that ensures predecessor operation being added did not skip any operations between op and pre.
```
Would be good to define op and pre here.

Reply:
Done!